crates/oauth2-types/src/oidc.rs

```diff
@@ -433,10 +483,18 @@ impl ProviderMetadata {
             validate_url("introspection_endpoint", url, ExtraUrlRestrictions::None)?;
         }
 
+        let introspection_methods = metadata
+            .introspection_endpoint_auth_methods_supported
+            .as_ref()
+            .map(|v| {
+                v.iter()
+                    .filter_map(AuthenticationMethodOrAccessTokenType::authentication_method)
+                    .collect::<Vec<_>>()
+            });
```

Comment:
So basically here you're doing that to make it a Vec of only the auth methods, right? Could you just add a comment saying that the introspection methods can also include token types, and that's why you have to do that transformation first?